src/test/java/net/snowflake/client/jdbc/BindingDataLatestIT.java

```diff
@@ -122,57 +136,189 @@ public void testTimestampBindingWithNTZType() throws SQLException {
+   * Both regular binding and stage binding have such issue but they also behave
+   * diffently, for some data only regular binding gets the extra hour while
+   * sometime only stage binding does.
+   * The workaround is to use CLIENT_TIMESTAMP_TYPE_MAPPING=LTZ to insert
+   * LTZ/TZ data and use CLIENT_TIMESTAMP_TYPE_MAPPING=NTZ to insert NTZ data.
+   *
+   * @throws SQLException
+   */
+  @Test
+  @ConditionalIgnoreRule.ConditionalIgnore(condition = RunningOnGithubAction.class)
+  public void testTimestampBindingWithLTZTypeForDayLightSavingTimeZone() throws SQLException {
+    TimeZone origTz = TimeZone.getDefault();
+    Calendar australia = Calendar.getInstance(TimeZone.getTimeZone("Australia/Sydney"));
+    TimeZone.setDefault(TimeZone.getTimeZone("Australia/Sydney"));
+    try (Connection connection = getConnection();
+        Statement statement = connection.createStatement()) {
+      try {
+        statement.execute(
+            "create or replace table stageinsert(ind int, ltz0 timestamp_ltz, ltz1 timestamp_ltz, ltz2 timestamp_ltz, tz0 timestamp_tz, tz1 timestamp_tz, tz2 timestamp_tz, ntz0 timestamp_ntz, ntz1 timestamp_ntz, ntz2 timestamp_ntz)");
+        statement.execute(
+            "create or replace table regularinsert(ind int, ltz0 timestamp_ltz, ltz1 timestamp_ltz, ltz2 timestamp_ltz, tz0 timestamp_tz, tz1 timestamp_tz, tz2 timestamp_tz, ntz0 timestamp_ntz, ntz1 timestamp_ntz, ntz2 timestamp_ntz)");
+        statement.execute("alter session set CLIENT_TIMESTAMP_TYPE_MAPPING=TIMESTAMP_LTZ");
+        statement.execute("alter session set TIMEZONE='UTC'");
+
+        Timestamp ts1 = new Timestamp(1403049600000L);
+        Timestamp ts2 = new Timestamp(1388016000000L);
+        Timestamp ts3 = new Timestamp(System.currentTimeMillis());
+
+        // insert using regular binging
+        try (PreparedStatement prepStatement =
+            connection.prepareStatement("insert into regularinsert values (?,?,?,?,?,?,?,?,?,?)")) {
+          for (int i = 1; i <= 6; i++) {
+            prepStatement.setInt(i, 1);
+            prepStatement.setTimestamp(2, ts1);
+            prepStatement.setTimestamp(3, ts2);
+            prepStatement.setTimestamp(4, ts3);
+
+            prepStatement.setTimestamp(5, ts1);
+            prepStatement.setTimestamp(6, ts2);
+            prepStatement.setTimestamp(7, ts3);
+
+            prepStatement.setTimestamp(8, ts1, australia);
+            prepStatement.setTimestamp(9, ts2, australia);
+            prepStatement.setTimestamp(10, ts3, australia);
+
+            prepStatement.addBatch();
+          }
+          prepStatement.executeBatch();
+        }
+
+        // insert using stage binding
+        try (PreparedStatement prepStatement =
+            connection.prepareStatement("insert into stageinsert values (?,?,?,?,?,?,?,?,?,?)")) {
+          statement.execute("ALTER SESSION SET CLIENT_STAGE_ARRAY_BINDING_THRESHOLD = 1");
```

Comment:
let's move the alter session before try and reuse code executing prepared statement

Reply:
Done.